### DISCUSSION TITLE
Georgios' feedback

### DISCUSSION THREAD

**Discussion by gkaramanis** [EvaMaeRey/easy-geom-recipes#4] [General]
The format works for me. I've read tutorials about creating extensions before but this one made it look easier and more intuitive. I want to try again writing my own! 

I would prefer the steps to be visually more separated, maybe add  some space between them? I felt a bit overwhelmed in a couple places between chapters with the code blocks, callouts and big headers close to each other.

Some sentences were difficult for me to follow, I had to read a second or third time, like the pro tip in step 0 and the callout in step 2 ("Inputting appropriate…").

"You" and "we" language. Personally I don't mind using both but maybe better to be consistent?

**Comment by EvaMaeRey** [comment]
I think the feedback is really on point. Standardize language (you/we), work on language clarity and simplicity ('Inputting appropriate...' is sounding awful! Sorry!), visual hierarchy and un-overwhelming.  I think this is tough regarding overwhelm.  I want people to have the sense that the callouts are something they can come back to. I tend to think, it's kind of best to skip/skim that information entirely the first time through, (especially because my verbalization (especially as is) of what you already may be noticing isn't necessarily that helpful and may take away from the big picture; and if you do need the clarification, seeing the big picture and building footholds might be key to understand the discussion). Did you end up reading all the tips and notes as you went?  I'll look into adding space/line between section as a starting point.   Thanks so much!

**Reply by gkaramanis** [reply]
I read most of the callouts, I skimmed one that I thought was a bit too "technical" for me. I think it was a great call to "hide" some information, it really makes the tutorial easier to follow. My feeling after reading it was that it was a very short one (in a good way, like easy to come back to), even though it's actually a lot of information when you unpack all the tips and notes.

The thing with the space or visual hierarchy might be a me thing, I know I have a strong preference for that 😄  (plus I was quite tired when I was reading the tutorial)

**Reply by EvaMaeRey** [reply]
Thanks for this additional detail.  I think your visual instincts are pretty good (data viz star level) and resonate with me, re hierarchy.  Probably most might feel the same way, but might not articulate it?

Here's a re-write idea for for 'inputting appropriate... 'Specifying `required_aes` for your Stat can improve your user interface; when used in plotting and a required aes isn't mapped, a standard error messages.'  I'm thinking I could include an example of this too, code actually returning the error message or just quoting the error message -- which, if you're experienced ggplot2 user you've probably seen a few times :-)

**Reply by EvaMaeRey** [reply]
@gkaramanis I saw a recent waterfall chart you made - olympic medals.  https://twitter.com/geokaramanis/status/1840341232754602100 🤩 If you are interested and have time, I wonder if you'd mind taking a look https://evamaerey.github.io/ggplot2-extension-cookbook/#geom_waterfall-111-compute_panel-geomrect  
The 'cookbook' is a collection of examples and intended follow-on to 'easy recipes' - much more underdeveloped in terms of form.  The waterfall is actually the first 'compute_panel' example - conceptually new territory, which can be trickier to manage.  I'd be curious how much the introduction resonates with you or not in terms of explaining why group-wise compute is insufficient.  Just thought I'd ask since you recently had thought through the waterfall computation requirements and because your feedback has been so helpful for recipes. 🙌

**Reply by EvaMaeRey** [reply]
I read the intro and I buryed the lead. So re-writing.

> A typical waterfall plot displays inflows and outflows that occur as a result as the *net change* across multiple events on the x axis.  Defining group-wise computation with compute_group, which we've done thus far, is not a good fit for this problem.  This is because we want to track the net change *across* the x variable, events, and because we anticipate events to generally be discrete.  Defining compute_group would indicate that the data should be divided by the x-axis events categories and compute done within each group instead of across groups.  So, we will define panel-wise computation with compute_panel.  

Before, the compute_group v compute_panel discussion wasn't at the forefront. 

> A waterfall plot displays inflows and outflows that occur as a result of events as well as the balance across multiple events. It is typically displayed as a series of rectangles. Because the net change is displayed (cumulative change), there is interdependence between the geometries on our plot – where one rectangle ends, the next in the series begins. Therefore we’ll be computing by panel and not by group – we do not want ggplot2 to split the data by discrete variables, which our x axis is most likely to be.

**Comment by gkaramanis** [comment]
Sorry for the very late reply, I was swamped with work and other stuff last week!
 
 I liked the flow of the tutorial. Although I didn't get all the technical details, it was really easy to follow! And I liked how it goes from simple to more complex.
 
 Just a couple small things: 

> Because the net change is displayed (cumulative change)

 I'm not sure _cumulative change_ helps here, I'm not sure but I understand it as a different thing than _net change_?
 <br> 
> we do not want ggplot2 to split the data by discrete variables, which our x axis is most likely to be. 

Not obvious for me why we do not want that, maybe move here the reason you mention later: _Group-wise computation (the defining compute_group element), would fail us, as the cross-event interdependence would not be preserved._
 <br>
> For ‘step 0’, we base ggplot2

Is there a missing verb here? Like, we _use_ base ggplot2?
 <br> 
> maybe add factor in order if factor is not defined...

I see the comment is incomplete but I'm not sure I understand this

**Reply by EvaMaeRey** [reply]
Awesome feedback!  I will work on incorporating and comment back.  Thanks!!
